x-pack/test/api_integration/deployment_agnostic/apis/observability/ai_assistant/distributed_lock_manager/distributed_lock_manager.spec.ts

```diff
@@ -0,0 +1,493 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { v4 as uuid } from 'uuid';
+import {
+  LockId,
+  ensureTemplatesAndIndexCreated,
+  LockManager,
+  withLock,
+  LockDocument,
+  LOCKS_CONCRETE_INDEX_NAME,
+} from '@kbn/observability-ai-assistant-plugin/server/service/distributed_lock_manager';
+import { Client } from '@elastic/elasticsearch';
+import { times } from 'lodash';
+import { ToolingLog } from '@kbn/tooling-log';
+import type { DeploymentAgnosticFtrProviderContext } from '../../../../ftr_provider_context';
+import { getLoggerMock } from '../utils/logger';
+
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
```

Comment:
you can also use jest integration tests for this, might be easier because you don't need to spin up Kibana.

Reply:
Ah! That might be a better fit. Will they then run against my local ES (:9200) when running `yarn test:jest_integration`?

Reply:
I don't know, but it's a good question, and for sure it would be useful

Reply:
Asked in https://elastic.slack.com/archives/C5UDAFZQU/p1743512184191419

Reply:
https://github.com/elastic/kibana/issues/216690

Reply:
I'll leave the tests where they are for now. It's super handy being able to run them without restarting ES every time. I'll move them over as soon as https://github.com/elastic/kibana/issues/216690 is addressed